rllib/env/tests/test_single_agent_episode.py

```diff
@@ -0,0 +1,303 @@
+        for i in range(100):
+            action = i
+            obs, reward, is_terminated, is_truncated, _ = env.step(action)
+            episode_1.add_timestep(
+                observation=obs,
+                action=action,
+                reward=reward,
+                is_terminated=is_terminated,
+                is_truncated=is_truncated,
+            )
+
+        # Assert that the episode has indeed 100 timesteps.
+        self.assertTrue(episode_1.t == 100)
+
+        # Create a successor.
+        episode_2 = episode_1.create_successor()
+        # Assert that it has the same id.
+        self.assertTrue(episode_1.id_ == episode_2.id_)
+        # Assert that the timestep starts at the end of the last episode.
+        self.assertTrue(episode_1.t == episode_2.t == episode_2.t_started)
+        # Assert that the last observation of `episode_1` is the first of
+        # `episode_2`.
+        self.assertTrue(episode_1.observations[-1] == episode_2.observations[0])
+
+    def test_concat_episode(self):
+        # Create two episodes and fill them with 100 timesteps each.
+        env = TestEnv()
+        init_obs, _ = env.reset()
+        episode_1 = SingleAgentEpisode()
+        episode_1.add_initial_observation(initial_observation=init_obs)
+        # Sample 100 timesteps.
+        for i in range(100):
+            action = i
+            obs, reward, is_terminated, is_truncated, _ = env.step(action)
+            episode_1.add_timestep(
+                observation=obs,
+                action=action,
+                reward=reward,
+                is_terminated=is_terminated,
+                is_truncated=is_truncated,
+            )
+
+        # Create an successor.
+        episode_2 = episode_1.create_successor()
+
+        # Now, sample 100 more timesteps.
+        for i in range(100, 200):
+            action = i
+            obs, reward, is_terminated, is_truncated, _ = env.step(action)
+            episode_2.add_timestep(
+                observation=obs,
+                action=action,
+                reward=reward,
+                is_terminated=is_terminated,
+                is_truncated=is_truncated,
+            )
+
+        # Assert that the second episode's `t_started` is at the first episode's
+        # `t`.
+        self.assertTrue(episode_1.t == episode_2.t_started)
+        # Assert that the second episode's `t` is at 200.
+        self.assertTrue(episode_2.t == 200)
+
+        # Concate the episodes.
+        episode_1.concat_episode(episode_2)
+        # Assert that the concatenated episode start at `t_started=0`
+        # and has 200 sampled steps, i.e. `t=200`.
+        self.assertTrue(episode_1.t_started == 0)
+        self.assertTrue(episode_1.t == 200)
+        # Assert that all lists have appropriate length.
+        self.assertTrue(
+            len(episode_1.actions)
+            == len(episode_1.rewards)
+            == len(episode_1.observations) - 1
+            == 200
+        )
```

Comment:
Can we quickly assert here that `episode_2` is untouched after this concatenation and also check some of the indices (like assert that `episode_1.observations[105]` is the equal (but not the same object) as `episode_2.observations[5]`)?

Reply:
Great idea @sven1977 ! I will implement this!

Reply:
I think you are right. It's impossible for episode_2's data objects to be used in the now elongated episode_1.
Ok, fine with me. Let's remove these comments here and that's it :)